daemon/tests/happy_path.rs

```diff
@@ -86,28 +71,34 @@ async fn taker_takes_order_and_maker_rejects() {
     assert!(matches!(maker_cfd.state, CfdState::Rejected { .. }));
 }
 
+// Helper function setting up a "happy path" wallet mock
+fn mock_wallet_transaction(wallet: &mut MutexGuard<'_, harness::mocks::wallet::MockWallet>) {
+    #[allow(clippy::redundant_closure)] // clippy is in the wrong here
+    wallet
```

Comment:
I am not sure this helper function is a good thing. We can do this, but I don't think `build_party_params` should be part of it - build party param is specific to the contract setup. I think combining `sign` and `txid` makes sense, because we sign and send a transaction for many cases (i.e. commit, ...).

Maybe we can keep this code explicitly in the test and extract helper functions a bit later, once we have a better idea of what makes sense to combine.

Reply:
happy to inline this helper, I just didn't want to make this test unnecessarily verbose in my opinion. It doesn't make any difference in the behaviour.

Reply:
ended up removing build_party_params() from this function.

I also added a sequence, that means that these handlers have to be invoked in that order.